### PR TITLE
Add XML Namespace support for appendNewLine to FileWritingMessageHandler (See INT-3621)

### PR DESCRIPTION
**Pull request by tonyfalabella**
I've added XML Namespace support for `appendNewLine(boolean)` to `FileWritingMessageHandler` to be considered for Spring Integration 4.2.  See https://jira.spring.io/browse/INT-3621 

Note that the attribute in the XML is named `append-new-line` which is slightly different than what it is called for the `CharacterStreamWritingMessageHandler` (that being `append-newline`).  The reason for the difference is that it must match the setter that was chosen on the class:  `FileWritingMessageHandler.setAppendNewLine(boolean)`.

/CC @artembilan @garyrussell

### CODE REVIEW

**Comment by artembilan** [comment]
@tonyfalabella , your work for the `appendNewLine` is merged. 
Would be better, if you rebase this branch to the latest `master` state.
From other side, please, follow with Contribution Guide (https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.md) how to form the first PR's commit.

I'll share with you some comments to your Namespace support commit.

File: spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.2.xsd [hunk 1]
```diff
@@ -349,105 +349,125 @@ Only files matching this regular expression will be picked up by this adapter.
...
-            </xsd:annotation>
-        </xsd:attribute>
-		<xsd:attributeGroup ref="integration:smartLifeCycleAttributeGroup"/>
+    			</xsd:documentation>
+    		</xsd:annotation>
+    	</xsd:attribute>
+    	<xsd:attribute name="append-new-line" type="xsd:string" default="false">
+    	    <xsd:annotation>
+    			<xsd:documentation>
+    				<![CDATA[Set to 'true' to append a new-line after each write (only applicable if mode='APPEND').
```
Comment (artembilan) [inline spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.2.xsd]: `only applicable...` isn't relevant anymore. See the final solution from the related PR.
And, please, revert any other formatting changes. It's very difficult to determine exactly your changes.

File: spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests.java [hunk 2]
```diff
@@ -194,6 +211,44 @@ public void adapterUsageWithAppend() throws Exception{
...
+			testFile.delete();
+		}
+		adapterUsageWithAppendAndAppendNewLineFalse.send(new GenericMessage<String>("Initial File Content:"));
+		adapterUsageWithAppendAndAppendNewLineFalse.send(new GenericMessage<String>("String content:"));
+		adapterUsageWithAppendAndAppendNewLineFalse.send(new GenericMessage<byte[]>("byte[] content:".getBytes()));
+		adapterUsageWithAppendAndAppendNewLineFalse.send(new GenericMessage<File>(new File("test/input.txt")));
+
+		String actualFileContent = new String(FileCopyUtils.copyToByteArray(testFile));
+		assertEquals(expectedFileContent, actualFileContent);
+		testFile.delete();
```
Comment (artembilan) [inline spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests.java]: ... and `@author` to this class.

**Comment by artembilan** [comment]
If you won't mind, would be great to add to the `whats-new.xml` a note on the matter. As well as to the `file.xml` somewhere in the `file-writing` section.

File: spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests-context.xml [hunk 3]
```diff
@@ -52,7 +57,21 @@
...
+								   append-new-line="true"
+								   directory-expression="@barString">	
+	</file:outbound-channel-adapter>
 
+	<file:outbound-channel-adapter id="adapterUsageWithAppendAndAppendNewLineFalse"
+	                               filename-generator-expression="@fooString"
+								   mode="APPEND"
+								   append-new-line="false"
+								   directory-expression="@barString">	
+	</file:outbound-channel-adapter>
```
Comment (artembilan) [inline spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests-context.xml]: BTW, the `<int-file:outboud-gateway>` should support this atribute as well. According to the `outboundFileBaseType` XSD type it is already here. But we should have at least one test for that component to be sure parser populates it correctly. No need those integration tests as you did for the `FileOutboundChannelAdapterParserTests`

**Comment by artembilan** [comment]
@tonyfalabella , that's all. Looks straightforward, but needs those polishing. 
Thanks

File: spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.2.xsd (hunk 1 above)
Comment (tonyfalabella) [inline spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.2.xsd]: Ok, I'll take a look at this (realistically I won't be able to look at this until Weds night).  

Yes the diffs to the XML look like many changes were made.  Was unclear why the diff-tool did that as I didn't reformat the XML in my workspace.  Will see if I can figure out why it did that too.

File: spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests-context.xml (hunk 3 above)
Comment (tonyfalabella) [inline spring-integration-file/src/test/java/org/springframework/integration/file/config/FileOutboundChannelAdapterParserTests-context.xml]: Ok I'll add tests for `<int-file:outboud-gateway>`

**Comment by artembilan** [comment]
@tonyfalabella , I see you did some work here. But It looks like done wrong way. 
You should **fetch** the latest `master` and **rebase** this branch, not **merge**.
Or.. just create a new fresh branch and raise new PR

**Comment by tonyfalabella** [comment]
Yes I had all the changes ready but had issues with the *rebase* and merge.  So as you saw what's in this PR is messy now.  I'll probably just open a new PR when I get home tonight.

**Comment by tonyfalabella** [comment]
@artembilan I think I've addressed the things you asked me to change.  

To fix my branch I ended up *reset*ing my branch.  In doing so I think I may have taken it back to just prior to the mods you made for https://jira.springsource.org/browse/INT-3620.  I think I reapplied them properly, but you might want to cherry-pick all the files except for these two: `FileWritingMessageHandler` and `FileWritingMessageHandlerTests`

**Comment by artembilan** [comment]
@tonyfalabella , don't waste time trying fix this branch!

Just start a new one (e.g. `INT-3621a`) based on the `master` and copy/paste your changes to the namespace support from this branch. Finally you just raise a new PR and close this one as invalid.

**Comment by tonyfalabella** [comment]
@artembilan As you suggested I actually tried to create a fresh branch (the `INT-3621a` one you've noted) but the changes I made showed no differences when compared to the `origin`.  I'm pretty new to git so it was unclear if the commands I had done earlier messed up my local fork on github.  I was going to just whack the entire forked repos on github and start fresh (I have my changes on my local machine) but thought then that we'd lose this conversation as well with the items you wanted modified.

If you have any more suggestions that might be helpful.

**Comment by artembilan** [comment]
OK. Don't worry! It's just a process to used to work with.
Try to follow with this steps: https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.md
```
git fetch --all
git checkout master
git pull upstream master
git push origin master
git checkout -b INT-3621a
```
And that new branch will based on the current `master`.
After that you can compare your old `INT-3621` with this one to move changes with some `gitdiff` tool.

**Comment by tonyfalabella** [comment]
Closing this as it's replaced with PR pointing to branch INT-3621b

**Pull request closed without merging**